core/src/providers/openai.rs

```diff
@@ -799,6 +798,21 @@ impl LLM for OpenAILLM {
                 ))?,
             },
         }
+        self.use_eu_endpoint = match credentials.get("OPENAI_USE_EU_ENDPOINT") {
+            Some(use_eu_endpoint_str) => use_eu_endpoint_str == "true",
+            None => match tokio::task::spawn_blocking(|| std::env::var("OPENAI_USE_EU_ENDPOINT"))
+                .await?
+            {
+                Ok(use_eu_endpoint_str) => use_eu_endpoint_str == "true",
+                Err(_) => false,
+            },
+        };
+        self.host = match self.use_eu_endpoint {
```

Comment:
nit: using an `if` is more idiomatic than `match true/false`

Reply:
Darn, I thought I was being a good Ruster with match! I'll change it :)

Reply:
Not Ruster, Rustacean, I read that somewhere!